exaudfclient/base/javacontainer/script_options/test/ctpg_script_importer_test.cc

```diff
@@ -0,0 +1,57 @@
+
+#include "include/gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "base/javacontainer/script_options/parser_ctpg.h"
+
+#include "base/javacontainer/script_options/test/swig_factory_test.h"
+#include <string.h>
+
+using namespace SWIGVMContainers::JavaScriptOptions;
+
+
+static const char* sc_scriptName = "script";
+
+
+void checkIndex(size_t currentIdx, const char* scriptKey) {
+    std::stringstream ss;
+    ss << sc_scriptName << currentIdx;
+    if (ss.str() != scriptKey) {
+        throw std::logic_error(std::string("Script Key does not match: '") + ss.str() + " != '" + scriptKey + "'");
+    }
+}
+
+const char* buildNewScriptCode(size_t currentIdx) {
+    std::stringstream ss;
+    ss << "%import " << sc_scriptName << currentIdx << ";something";
+    static std::string ret;
+    ret = ss.str();
+    return ret.c_str();
+}
+
+TEST(ScriptImporterTest, max_recursion_depth) {
+
+    size_t currentIdx = 0;
+    SwigFactoryTestImpl swigFactoryTest([&](const char* scriptKey) {
```

Comment:
Nasty, you generate the Code when. It gets requested. Took a moment to see where this runs in a loop. Maybe a comment can help

Reply:
Added a comment